File: 1297_maximum_number_of_occurrences_of_a_substring/Solution.cs
// LeetCode 1297 - Maximum Number of Occurrences of a Substring
// https://leetcode.com/problems/maximum-number-of-occurrences-of-a-substring/

using System.Collections.Generic;

public class Solution {
    public int MaxFreq(string s, int maxLetters, int minSize, int maxSize) {
        var counts = new Dictionary<string, int>();
        for (int i = 0; i + minSize <= s.Length; i++) {
            string sub = s.Substring(i, minSize);
            var seen = new HashSet<char>();
            foreach (char ch in sub) seen.Add(ch);
            if (seen.Count <= maxLetters) {
                if (!counts.ContainsKey(sub)) counts[sub] = 0;
                counts[sub]++;
            }
        }
        int best = 0;
        foreach (var kv in counts) best = System.Math.Max(best, kv.Value);
        return best;
    }
}

File: 1297_maximum_number_of_occurrences_of_a_substring/solution.ts
// LeetCode 1297 - Maximum Number of Occurrences of a Substring
// https://leetcode.com/problems/maximum-number-of-occurrences-of-a-substring/

function maxFreq(s: string, maxLetters: number, minSize: number, maxSize: number): number {
    const counts = new Map();
    for (let i = 0; i + minSize <= s.length; i++) {
        const sub = s.slice(i, i + minSize);
        if (new Set(sub).size <= maxLetters) {
            counts.set(sub, (counts.get(sub) || 0) + 1);
        }
    }
    let best = 0;
    for (const freq of counts.values()) best = Math.max(best, freq);
    return best;
}
